File: simter-id-data/src/main/java/tech/simter/id/generator/impl/UUIDIdGenerator.java
package tech.simter.id.generator.impl;

import tech.simter.id.generator.IdGenerator;

import java.util.UUID;

/**
 * A random UUID id generator.
 * <p>
 * It's thread-safe for distribute system.
 *
 * @author RJ
 */
public class UUIDIdGenerator implements IdGenerator<UUID> {
  @Override
  public UUID nextId() {
    return UUID.randomUUID();
  }
}

File: simter-id-core/src/main/kotlin/tech/simter/id/generator/impl/UUIDIdGenerator.kt
package tech.simter.id.generator.impl

import tech.simter.id.IdGenerator

import java.util.UUID

/**
 * A random UUID id generator.
 *
 *
 * It's thread-safe for distribute system.
 *
 * @author RJ
 */
class UUIDIdGenerator : IdGenerator<UUID> {
  override fun nextId(): UUID {
    return UUID.randomUUID()
  }
}